daemons/fenced/fenced_history.c

```diff
@@ -240,6 +282,11 @@ stonith_merge_in_history_list(GHashTable *history)
             continue; /* skip over existant - state-merging migh be desirable */
         }
 
+        if (g_hash_table_size(stonith_remote_op_list) >= MAX_STONITH_HISTORY) {
```

Comment:
Rather than count the table every time, it would be more efficient to count it once at the beginning, and increment the count for each merged op. Or even wait until after the loop to check it once, though that means we might briefly go over the limit.

Reply:
That was my initial approach as well but looking at the implementation I realized that the counter is already part of the table so that it is cheap and thus not worth risking my private counter to get out of sync.

Reply:
Ah, I didn't realize GHashTable kept a counter. Guess I was thinking of GList

Reply:
Considering a smarter trim it probably makes sense to go over the limit and trim then. Anyway there is no significant additional memory-consumption as it is just moving ops between two hash-tables.